tls/extensions/s2n_npn.c

```diff
@@ -79,3 +79,62 @@ const s2n_extension_type s2n_server_npn_extension = {
     .should_send = s2n_server_npn_should_send,
     .if_missing = s2n_extension_noop_if_missing,
 };
+
+bool s2n_npn_encrypted_should_send(struct s2n_connection *conn)
+{
+    return s2n_server_alpn_should_send(conn);
+}
+
+S2N_RESULT s2n_calculate_padding(uint8_t protocol_len, uint8_t *padding_len)
+{
+    RESULT_ENSURE_REF(padding_len);
+    *padding_len = 32 - ((protocol_len + 2) % 32);
+    return S2N_RESULT_OK;
+}
+
+int s2n_npn_encrypted_extension_send(struct s2n_connection *conn, struct s2n_stuffer *out)
```

Comment:
why do some functions return `S2N_RESULT` while some return `int`?

Reply:
The definition of the [s2n_extension_type struct](https://github.com/aws/s2n-tls/blob/main/tls/extensions/s2n_extension_type.h#L35) expects its send/recv function pointers to return int. Helper methods in this file don't have to follow this rule, which is why the calculate_padding function returns S2N_RESULT.